onyo/lib/onyo.py

```diff
@@ -296,49 +329,56 @@ def _get_assets(self) -> set[Path]:
 
     def _get_dirs(self) -> set[Path]:
         """
-        Return a set of all directories in the repository (except under .git).
+        Return a set of all absolute `Path`s to directories in the repository
+        (except under .git).
         """
         log.debug('Acquiring list of directories')
-        dirs = {x.relative_to(self.root) for x in Path(self.root).glob('**/')
+        dirs = {self.sanitize_path(x) for x in Path(self.root).glob('**/')
                 if '.git' not in x.parts and
                 not x.samefile(self.root)}
 
         return dirs
 
     def _get_files(self) -> set[Path]:
         """
-        Return a set of all files in the repository (except under .git).
+        Return a set of all absolute `Path`s to files in the repository
+        (except under .git).
         """
         log.debug('Acquiring list of files')
-        files = {Path(x) for x in self._git(['ls-files', '-z']).split('\0') if x}
+        files = {Path(self.root, x) for x in self._git(['-C', str(self.root), 'ls-files', '-z']).split('\0') if x}
         return files
 
     def _get_files_changed(self) -> set[Path]:
         """
-        Return a set of all unstaged changes in the repository.
+        Return a set of all absolute `Path`s to unstaged changes in the
+        repository.
         """
         log.debug('Acquiring list of changed files')
-        changed = {Path(x) for x in self._git(['diff', '-z', '--name-only']).split('\0') if x}
+        changed = {Path(self.root, x) for x in self._git(['-C', str(self.root), 'diff', '-z', '--name-only']).split('\0') if x}
         return changed
 
     def _get_files_staged(self) -> set[Path]:
         """
-        Return a set of all staged changes in the repository.
+        Return a set of all absolute `Path`s to staged changes in the
+        repository.
         """
         log.debug('Acquiring list of staged files')
-        staged = {Path(x) for x in self._git(['diff', '--name-only', '-z', '--staged']).split('\0') if x}
+        staged = {Path(self.root, x) for x in self._git(['-C', str(self.root), 'diff', '--name-only', '-z', '--staged']).split('\0') if x}
         return staged
 
     def _get_files_untracked(self) -> set[Path]:
         """
-        Return a set of all untracked files in the repository.
+        Return a set of all absolute `Path`s to untracked files in the
+        repository.
         """
         log.debug('Acquiring list of untracked files')
-        untracked = {Path(x) for x in self._git(['ls-files', '-z', '--others', '--exclude-standard']).split('\0') if x}
+        untracked = {Path(self.root, x) for x in self._git(['-C', str(self.root), 'ls-files', '-z', '--others', '--exclude-standard']).split('\0') if x}
```

Comment:
This may be for later when `opdir` is removed, but FTR:

I'd just make `_git` always call `subprocess.run` with `cwd=self.root` rather than passing `-C str(self.root)` everywhere.